File: clients/node/sdk/models/userUsage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserUsage = void 0;
class UserUsage {
    /**
     * Gets the active_halfyear property value. The active_halfyear property
     * @returns a int64
     */
    get active_halfyear() {
        return this._active_halfyear;
    }
    ;
    /**
     * Sets the active_halfyear property value. The active_halfyear property
     * @param value Value to set for the active_halfyear property.
     */
    set active_halfyear(value) {
        this._active_halfyear = value;
    }
    ;
    /**
     * Gets the active_month property value. The active_month property
     * @returns a int64
     */
    get active_month() {
        return this._active_month;
    }
    ;
    /**
     * Sets the active_month property value. The active_month property
     * @param value Value to set for the active_month property.
     */
    set active_month(value) {
        this._active_month = value;
    }
    ;
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    get additionalData() {
        return this._additionalData;
    }
    ;
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    set additionalData(value) {
        this._additionalData = value;
    }
    ;
    /**
     * Instantiates a new UserUsage and sets the default values.
     */
    constructor() {
        this._additionalData = {};
    }
    ;
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    getFieldDeserializers() {
        return {
            "active_halfyear": n => { this.active_halfyear = n.getNumberValue(); },
            "active_month": n => { this.active_month = n.getNumberValue(); },
            "total": n => { this.total = n.getNumberValue(); },
        };
    }
    ;
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    serialize(writer) {
        if (!writer)
            throw new Error("writer cannot be undefined");
        writer.writeNumberValue("active_halfyear", this.active_halfyear);
        writer.writeNumberValue("active_month", this.active_month);
        writer.writeNumberValue("total", this.total);
        writer.writeAdditionalData(this.additionalData);
    }
    ;
    /**
     * Gets the total property value. The total property
     * @returns a int64
     */
    get total() {
        return this._total;
    }
    ;
    /**
     * Sets the total property value. The total property
     * @param value Value to set for the total property.
     */
    set total(value) {
        this._total = value;
    }
    ;
}
exports.UserUsage = UserUsage;

File: clients/csharp/sdk/Models/UserUsage.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System;
namespace MastodonClientLib.Models {
    public class UserUsage : IParsable {
        /// <summary>The active_halfyear property</summary>
        public long? ActiveHalfyear { get; set; }
        /// <summary>The active_month property</summary>
        public long? ActiveMonth { get; set; }
        /// <summary>The total property</summary>
        public long? Total { get; set; }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static UserUsage CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new UserUsage();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"active_halfyear", n => { ActiveHalfyear = n.GetLongValue(); } },
                {"active_month", n => { ActiveMonth = n.GetLongValue(); } },
                {"total", n => { Total = n.GetLongValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteLongValue("active_halfyear", ActiveHalfyear);
            writer.WriteLongValue("active_month", ActiveMonth);
            writer.WriteLongValue("total", Total);
        }
    }
}
